dashboard/serializers.py

```diff
@@ -38,6 +39,29 @@ def serialize_enrollments(cls, mmtrack, enrollments):
         return serialized_enrollments
 
     @classmethod
+    def count_courses_passed(cls, program, mmtrack):
+        """
+        Returns number of passed courses.
+
+        Args:
+            program (Program): A program object.
+            mmtrack (dashboard.utils.MMTrack): a instance of all user information about a program
+        Returns:
+            num_complete_courses (int): A number of passed courses.
+        """
+        num_courses_passed = 0
+        for course in program.course_set.all():
+            course_info_dict = get_info_for_course(course, mmtrack)
+            if course_info_dict:
+                expected_statuses = list(
+                    filter(lambda c: c['status'] in [CourseStatus.PASSED], course_info_dict["runs"])
+                )
+                if len(expected_statuses) > 0:
+                    num_courses_passed += 1
+
+        return num_courses_passed
```

Comment:
@noisecapella continuing the discussion from [here](https://github.com/mitodl/micromasters/pull/2133#discussion_r92888017)...

first of all i would stress that my suggestion was not a refactor. it was a change to some existing code to support a new feature. second, if my suggestion seemed heavy, i probably didn't explain myself well enough. here is the diff to implement my suggestion in its entirety: https://gist.github.com/gsidebo/812ecb6480656d772bbc85eb83217e8b

here are the diff stats for my changes on top of the current branch. i disagree that this is in any way heavy:
```
 dashboard/serializers.py | 30 +-----------------------------
 dashboard/utils.py       | 17 ++++++++++++++---
 2 files changed, 15 insertions(+), 32 deletions(-)
```

i tested locally and it works as expected. your approach uses `get_info_for_course`, which is used to serialize a course and all of its course runs. the work to loop through and serialize course runs is unnecessary for the purposes of counting the number of passed courses. you'll also see that my suggestion uses `MMTrack.has_passed_course`, which is the exact method that `get_info_for_course` uses to determine if a course is passed ([here](https://github.com/mitodl/micromasters/blob/master/dashboard/api.py#L247) and [here](https://github.com/mitodl/micromasters/blob/master/dashboard/api.py#L263)). `get_info_for_course` doesn't decide whether a course run is passed; `MMTrack.has_passed_course` does, and `get_info_for_course` makes use of it.